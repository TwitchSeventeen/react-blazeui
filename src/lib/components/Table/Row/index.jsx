import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames';

function Row(props) {
  const {
    children, className, head, disabled, clickable, ...other
  } = props;

  const classes = classNames(
    'c-table__row',
    { 'c-table__row--heading': head },
    { 'c-table__row--disabled': disabled },
    { 'c-table__row--clickable': clickable },
    className,
  );

  return (
    <tr className={classes} {...other}>
      { children }
    </tr>
  );
}

Row.propTypes = {
  /**
   * This element can have children
   */
  children: PropTypes.node,
  /**
   * Additional classes
   */
  className: PropTypes.string,
  /**
   * Defines this row is a heading
   */
  head: PropTypes.bool,
  /**
   * Defines a disabled row
   */
  disabled: PropTypes.bool,
  /**
   * Defines a clickable row
   */
  clickable: PropTypes.bool,
};

Row.defaultProps = {
  className: null,
  children: null,
  head: false,
  disabled: false,
  clickable: false,
};

export default Row;
