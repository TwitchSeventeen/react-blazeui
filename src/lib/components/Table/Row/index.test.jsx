import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Row from './index';

describe('Row', () => {
  it('should be defined', () => {
    expect(Row).toBeDefined();
  });

  it('renders correctly', () => {
    const renderer = new ShallowRenderer();
    renderer.render(
      <Row>
        lorem ipsum dolor
      </Row>,
    );

    const tree = renderer.getRenderOutput();
    expect(tree).toMatchSnapshot();
  });
});
