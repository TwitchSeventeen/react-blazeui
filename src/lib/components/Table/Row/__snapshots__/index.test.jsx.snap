// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Row renders correctly 1`] = `
<tr
  className="c-table__row"
>
  lorem ipsum dolor
</tr>
`;
